allennlp/nn/beam_search.py

```diff
@@ -293,18 +307,34 @@ def new_step(
             for key, state_tensor in state.items():
                 if state_tensor is None:
                     continue
-                _, *last_dims = state_tensor.size()
-                # shape: (batch_size, beam_size, *)
-                expanded_backpointer = backpointer.view(
-                    batch_size, self.beam_size, *([1] * len(last_dims))
-                ).expand(batch_size, self.beam_size, *last_dims)
-
-                # shape: (batch_size * beam_size, *)
-                state[key] = (
-                    state_tensor.reshape(batch_size, self.beam_size, *last_dims)
-                    .gather(1, expanded_backpointer)
-                    .reshape(batch_size * self.beam_size, *last_dims)
-                )
+                if multilayer_rnn_decoder:
```

Comment:
Where does this variable `multilayer_rnn_decoder` come from? It would need to be set for each `(key, state_tensor)` pair like it is above, right? Otherwise the value of `multilayer_rnn_decoder` here will just be equal to whatever it was last assigned above.

Reply:
Fix!